beetmoverscript/src/beetmoverscript/script.py

```diff
@@ -198,11 +200,29 @@ async def push_to_maven(context):
     context.checksums = dict()  # Needed by downstream calls
     context.raw_balrog_manifest = dict()  # Needed by downstream calls
 
+    check_maven_artifact_map(context)
+
     # overwrite artifacts_to_beetmove with the declarative artifacts ones
     context.artifacts_to_beetmove = task.get_upstream_artifacts(context, preserve_full_paths=True)
     await move_beets(context, context.artifacts_to_beetmove, artifact_map=context.task["payload"]["artifactMap"])
 
 
+def check_maven_artifact_map(context):
+    """Check that versions in artifact map to be valid Maven versions"""
+    version = context.task["payload"]["version"]
+    try:
+        MavenVersion.parse(version)
+    except PatternNotMatchedError:
```

Comment:
Good idea on matching a specific error. One thing I realize: `mozilla_version` raises another type of error: https://github.com/mozilla-releng/mozilla-version/blob/b541228cf29bd8286e0fbe9692a82bd39bb94804/mozilla_version/version.py#L142. It's not ideal because people unfamiliar with `mozilla-version` may except just a single exception - just like you did.

For now, let's use 

```suggestion
    except (ValueError, PatternNotMatchedError) as e:
```

but I believe your solution is what we want to have, in the end.

Reply:
After I applied your suggested change I realized that the `parse` method we are calling (https://github.com/mozilla-releng/mozilla-version/blob/b541228cf29bd8286e0fbe9692a82bd39bb94804/mozilla_version/version.py#L36) seems to only raise `PatternNotMatchedError` exception, or `MissingFieldError` due to `get_value_matched_by_regex`. I see that all the errors raised by `mozilla-version` are subclasses of `ValueError` anyway, so catching `ValueError` works too (and makes catching `PatternNotMatchedError` redundant, I think). 

Maybe it would be interesting to add something like a `ParsingError` exception class to `mozilla-version` that all related errors would inherit from? Such an exception is probably what I would expect as a consumer of the library, and it's a backwards-compatible change.